File: PRB01.py
import math

test_cases = int(input())

for _ in range(test_cases):
    inp = int(input())
    prime = True
    if inp == 1:
        print("no")
        continue
    elif inp == 2:
        print("yes")
        continue
    if inp % 2 == 0:
        print('no')
        continue
    for i in range(3, int(math.sqrt(inp))+1, 2):
        if inp % i == 0:
            prime = False
            break
        prime = True
    if prime:
        print('yes')
    else:
        print('no')

File: PRB01.go
package main

import (
	"fmt"
	"math"
)

func primeTest(n int) bool {
	if n == 1 {
		return false
	} else if n == 2 {
		return true
	}
	if n%2 == 0 {
		return false
	}
	sqrt := math.Floor(math.Sqrt(float64(n)))
	for i := 3; i < int(sqrt); i = i + 2 {
		if n%i == 0 {
			return false
		}
	}
	return true
}
func main() {
	var testCases, num int
	var res bool
	fmt.Scanf("%d", &testCases)
	for i := 0; i < testCases; i++ {
		fmt.Scanf("%d", &num)
		res = primeTest(num)
		if res {
			fmt.Println("yes")
		} else {
			fmt.Println("no")
		}
	}
}
